src/cpp/src/logit_processor.hpp

```diff
@@ -299,6 +299,20 @@ class PresencePenaltyTransform : public IPenaltyTransformer {
     }
 };
 
+
+class LogFilter : public ILogitTransformer {
+public:
+    LogFilter() {}
+    
+    void apply(Logits& logits) override {
+        OPENVINO_ASSERT(logits.is_vector_initialized());
```

Comment:
This is true only if we have top_p or top_k set. Without them we don't need to sort results hence new vector is not initialized. So it will not work if only temperature is provided. 
To make it work in both scenarios we need to change `logits.m_vector[i].m_log_prob` values **if vector is initialized** and 
`logits.m_data[i]` **if vector is not initialized**.